### ISSUE TITLE
bug: Cairo runner does not support array arguments larger than 2^15

### ISSUE THREAD

**Issue by m-kus** [starkware-libs/cairo#6479]
# Bug Report

**Cairo version:**

737cc88d14a17e1e8b46a6d2d4061aa4b6984980

**Current behavior:**

When running a program with an array argument that is of length 32768:
- In debug mode: Cairo runner fails with overflow when tries to increment the allocation pointer while loading arguments;
- In release mode: Cairo runner loads arguments without errors postponing the memory issue to the cairo-vm execution.

**Expected behavior:**

Cairo runner works without errors in both debug and release, there are no memory issues during the cairo-vm execution.

**Steps to reproduce:**

Unit test:
https://github.com/m-kus/cairo/pull/1/files

**Related code:**

The minimal program to reproduce the issue:

```rust
fn main(args: Array<felt252>) {}
```

**Other information:**

Possibly related:
- https://github.com/lambdaclass/cairo-vm/issues/1848#issuecomment-2397683850
- https://github.com/lambdaclass/cairo-vm/pull/1759

**Comment by maciejka** [comment]
When the arg size is large the `ap_offset` that is tracking `ap` overflows here: https://github.com/starkware-libs/cairo/blob/737cc88d14a17e1e8b46a6d2d4061aa4b6984980/crates/cairo-lang-runner/src/lib.rs#L864.

**Comment by orizi** [comment]
This should be done as part of https://github.com/lambdaclass/cairo-vm/pull/1759 - as the issue is the amount of instructions created for this explicit array creation.

Is going to be solved at the future when the code isn't actually changing by the specific inputs it received.

In the near future we should have a way of running code, where the input is *always* a `Span` of felts and the output is *always* a `Span` of felts, where both are directly deserialized (from the inputs) and serialized (into the output builtin) providing a provable method of run.

So basically - it isn't a bug - as this was just for local purposes (as `cairo-run` didn't actually run anything provable, so i would have expected it to work) - but the missing feature should exist soon enough.
